tensorflow_datasets/core/utils/tf_utils.py

```diff
@@ -62,6 +62,8 @@ class TFGraphRunner(object):
 
   """
 
+  __slots__ = ["_graph_run_cache"]
```

Comment:
Not in this PR, but a better optimization is probably to remove the `TFGraphRunner` entirely. I believe it is only used for encoding images, and currently it recreate a new graph each time an input shape is different.
I believe images could be encoded using a unique graph with `placeholder(shape=(None, None, channel))` (or tf.function(inputs=[TensorSpec(shape=(None,None,channel)]) in eager)).

Currently eager functions are not even wrapped in tf.function

Reply:
> Currently eager functions are not even wrapped in tf.function

Good point. Is there a timeline for dropping TF 1.x support in TFDS which would make such optimizations a lot easier?

Reply:
> Good point. Is there a timeline for dropping TF 1.x support in TFDS which would make such optimizations a lot easier?

Even after dropping TF 1.5 support, we're not planning to drop support for graph mode any time soon as it is still widely used (People can still use `tf.compat.v1.Session` in TF 2.x).
The easiest here would be to only implement graph mode wrapped in `tf.Graph` as it should work both for eager/graph.